src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt

```diff
@@ -134,3 +134,22 @@ implication        ,ARM64 ,Sha1      ,ArmBase
 implication        ,ARM64 ,Sha256    ,ArmBase
 implication        ,ARM64 ,Vector64  ,AdvSimd
 implication        ,ARM64 ,Vector128 ,AdvSimd
+
+
+;                    ,name and aliases           ,archs    ,lower baselines included by implication
+;
+instructionsetgroup  ,x86-x64                    ,X64 X86  ,sse2
+instructionsetgroup  ,x86-x64-v2                 ,X64 X86  ,sse4.2 popcnt aes
+instructionsetgroup  ,x86-x64-v3 skylake         ,X64 X86  ,x86-x64-v2 avx2 bmi2 lznct pclmul movbe fma
```

Comment:
BMI2 does not imply BMI1, they are distinct queries that do not depend on eachother:
```suggestion
instructionsetgroup  ,x86-x64-v3 skylake         ,X64 X86  ,x86-x64-v2 avx2 bmi1 bmi2 lznct pclmul movbe fma
```

`pclmul` and `aes` notably seem to be not included in the "official" list for `x86-x64-v*`. I wonder if this is due to the same "export restrictions" that ARM has and if there may be platforms its not available under?

Notably, `pclmul` was introduced at the same time as `AES` and is part of the "cryptographic" instructions. I don't see any CPUs that have `AES` that don't also have `pclmul`

Reply:
@tannergooding so what is your suggestion around them? to drop them from all -v3 and make skylake a separate line with them or leave as is?

also, it's `bmi` not `bmi1` 🙂

Reply:
> to drop them from all -v3 and make skylake a separate line with them or leave as is?

Given that the mailing list did not include `aes`, `sha1`, `sha256`, or `pclmulqdq` in the `x86-x64-v*` list, I think its appropriate for us to not include it either. Even the LLVM page (https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/Support/X86TargetParser.def#L222-L255) doesn't actually include `AES` in any of these lists, it only calls it out as part of the `core_aes_pclmulqdq` name but doesn't actually have a `+aes` or `+pclmul` anywhere.

Our own recommendation is also that most users don't use the Crypto ISAs. They should be using `System.Security.Cryptography` instead and so not having it enabled by default seems "good".

> also, it's bmi not bmi1 🙂

I don't mind if we support `bmi` for parity with LLVM, but `bmi1` is the official name, the name used by the runtime, the `DOTNET_Enable*` environment variables, and the docs so we should at a minimum support the official name as well and default to using it in our own things (for consistency, discoverability, clarity, etc).

Reply:
For reference, the official levels were documented here as part of the official Linux x86-x64 ABI documentation: https://gitlab.com/x86-psABIs/x86-64-ABI/-/blob/master/x86-64-ABI/low-level-sys-info.tex

Reply:
Addressed